worker/src/queues/ingestionQueue.ts

```diff
@@ -34,78 +35,106 @@ const getS3StorageServiceClient = (bucketName: string): StorageService => {
   return s3StorageServiceClient;
 };
 
-export const ingestionQueueProcessor: Processor = async (
-  job: Job<TQueueJobTypes[QueueName.IngestionQueue]>,
-) => {
-  try {
-    const span = getCurrentSpan();
-    if (span) {
-      span.setAttribute("messaging.bullmq.job.input.id", job.data.id);
-      span.setAttribute(
-        "messaging.bullmq.job.input.projectId",
-        job.data.payload.authCheck.scope.projectId,
-      );
-      span.setAttribute(
-        "messaging.bullmq.job.input.eventBodyId",
-        job.data.payload.data.eventBodyId,
-      );
-      span.setAttribute(
-        "messaging.bullmq.job.input.type",
-        job.data.payload.data.type,
+export const ingestionQueueProcessorBuilder = (
+  enableRedirectToSecondaryQueue: boolean,
+): Processor => {
+  const projectIdsToRedirectToSecondaryQueue =
+    env.LANGFUSE_SECONDARY_INGESTION_QUEUE_ENABLED_PROJECT_IDS?.split(",") ??
+    [];
+
+  return async (job: Job<TQueueJobTypes[QueueName.IngestionQueue]>) => {
+    try {
+      const span = getCurrentSpan();
+      if (span) {
+        span.setAttribute("messaging.bullmq.job.input.id", job.data.id);
+        span.setAttribute(
+          "messaging.bullmq.job.input.projectId",
+          job.data.payload.authCheck.scope.projectId,
+        );
+        span.setAttribute(
+          "messaging.bullmq.job.input.eventBodyId",
+          job.data.payload.data.eventBodyId,
+        );
+        span.setAttribute(
+          "messaging.bullmq.job.input.type",
+          job.data.payload.data.type,
+        );
+      }
+
+      if (
+        enableRedirectToSecondaryQueue &&
+        projectIdsToRedirectToSecondaryQueue.includes(
+          job.data.payload.authCheck.scope.projectId,
+        )
+      ) {
+        logger.debug(
+          `Redirecting ingestion event to secondary queue for project ${job.data.payload.authCheck.scope.projectId}`,
+        );
+        const secondaryQueue = getQueue(QueueName.IngestionSecondaryQueue);
+        if (secondaryQueue) {
+          await secondaryQueue.add(QueueName.IngestionSecondaryQueue, job.data);
+        }
+        return;
+        // If we don't redirect, we continue with the ingestion. Otherwise, we finish here.
```

Comment:
Consider returning within the if statement. That way we continue with normal processing in case the secondary queue cannot be initiated for any reason.